ansible/tasks/initdb.yml

```diff
@@ -22,7 +22,30 @@
     user: "{{ db_username }}"
     password: "{{ db_password }}"
     force_basic_auth: yes
-  when: result is defined and result.status == 404
+  when: dbexists is defined and dbexists.status == 404
+
+# fetches the revision of previous view (to update it) if it exists
+- name: check for previous view in "auth" database
+  vars:
+    auth_index: "{{ lookup('file', '{{ openwhisk_home }}/ansible/files/auth_index.json') }}"
+  uri:
+    url: "{{ db_protocol }}://{{ db_host }}:{{ db_port }}/{{ db_auth }}/{{ auth_index['_id'] }}"
+    return_content: yes
+    method: GET
+    status_code: 200, 404
+    user: "{{ db_username }}"
+    password: "{{ db_password }}"
+    force_basic_auth: yes
+  register: previousView
+  when: dbexists is defined and dbexists.status != 404 # and mode=="updateview"
```

Comment:
Are you concerned about accidentally running this against a large database in a production environment? My thought was that you would not run `initdb` again in production, other than to update the view.

There are two points:
1. always updating when these tasks run (i.e., without needing an `updateview` mode for example)
2. not updating if the new design doc is the same as the previous design doc modulo the revision.

You might be thinking of doing 2 as well? I'm good with that.